src/main/java/uk/gov/hmcts/reform/pip/account/management/service/AccountService.java

```diff
@@ -505,6 +505,30 @@ public PiUser getUserById(UUID userId) {
             "User with supplied user id: %s could not be found", userId)));
     }
 
+    public AzureAccount retrieveUser(String provenanceUserId) {
+        try {
+            Optional<PiUser> user = userRepository.findByProvenanceUserIdAndUserProvenance(
+                provenanceUserId, UserProvenances.PI_AAD);
+            if (user.isPresent()) {
+                AzureAccount azureUser = new AzureAccount();
+                User aadUser = azureUserService.getUser(user.get().getEmail());
+                azureUser.setAzureAccountId(aadUser.id);
+                azureUser.setFirstName(aadUser.givenName);
+                azureUser.setSurname(aadUser.surname);
+                azureUser.setDisplayName(aadUser.displayName);
+                azureUser.setEmail(user.get().getEmail());
+                return azureUser;
+            } else {
+                throw new NotFoundException(String.format(
+                    "User with supplied issuer id: %s could not be found", provenanceUserId));
```

Comment:
Issuer ID in the comment here should be changed to provenance user id

Reply:
fixed.